middleware/getResults.js

```diff
@@ -0,0 +1,41 @@
+const http = require('http');
+const async = require('async');
+const xmlParser = require('../lib/xmlParser');
+const utils = require('../lib/resultsUtils');
+const processResults = require('../lib/resultsProcessor.js');
+
+function getData(requestUrl, page, cb) {
+  const pagedRequestUrl = `${requestUrl}${page}`;
+
+  http.get(pagedRequestUrl, (res) => {
+    let xml = '';
+
+    res.on('data', (chunk) => {
+      xml += chunk;
+    });
+
+    res.on('end', () => {
+      // TODO: Handle responses that are not 200
+      const pharmacyList = xmlParser(xml);
+      cb(null, pharmacyList);
+    });
+  });
+}
+
+function getResults(req, res, next) {
+  const requestUrl = utils.generateRequestUrl(req);
+
+  const tenRequests = [];
+  [1, 2, 3, 4, 5, 6, 7, 8, 9, 10].forEach((item) => {
```

Comment:
I'm pretty sure there is a better way to generate an array of 10 functions but this works and is reasonably clear.